the-collab-lab/tcl-28-smart-shopping-list
# Sd input component

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



- Created `Input` reusable component using Material UI's `TextField`
- `Input` component takes in props (label, name, value, onChange, required) Thank you @isaabutaa for the amazing idea! 🤩
- Applied `Input` component to entering an existing token, filtering items, and adding a new item


## Related Issue

closes #47 



## Acceptance Criteria





## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before



<img width="624" alt="Screen Shot 2021-08-31 at 10 02 59 PM" src="https://user-images.githubusercontent.com/60532744/131599699-99f50e90-a244-4123-a532-2e2097a65a55.png">

<img width="425" alt="Screen Shot 2021-08-31 at 10 03 34 PM" src="https://user-images.githubusercontent.com/60532744/131599741-a1622dff-dc89-4cab-8ed6-6cae04305b65.png">

<img width="382" alt="Screen Shot 2021-08-31 at 10 03 39 PM" src="https://user-images.githubusercontent.com/60532744/131599747-9c5afba7-de4b-4d2d-a8c3-6e7b1cd0c3a8.png">


### After

<img width="581" alt="Screen Shot 2021-08-31 at 9 59 43 PM" src="https://user-images.githubusercontent.com/60532744/131599497-7738f213-3904-4493-8a6b-1831df59fbf9.png">

<img width="386" alt="Screen Shot 2021-08-31 at 10 00 00 PM" src="https://user-images.githubusercontent.com/60532744/131599504-3e7bac64-ac30-4090-a438-a0ce40ea0ddb.png">

<img width="486" alt="Screen Shot 2021-08-31 at 10 00 33 PM" src="https://user-images.githubusercontent.com/60532744/131599509-cf76ca45-c7b2-4625-9fbb-2e0b10f5a5ed.png">





## Testing Steps / QA Criteria

-  Pull down `sd-input-component`
- Checkout to `sd-input-component`
- Check to see if you are still able to enter an existing token using the new input
- Check to see if you are still able to filter properly using the new input
- Check to see if you are still able to add a new item using the new input
